src/coreclr/nativeaot/Runtime/PalRedhawkFunctions.h

```diff
@@ -79,21 +79,12 @@ inline HANDLE PalGetCurrentThread()
     return GetCurrentThread();
 }
 
-#ifdef UNICODE
-_Success_(return != 0 && return < nSize)
-extern "C" uint32_t __stdcall GetEnvironmentVariableW(_In_opt_ LPCWSTR lpName, _Out_writes_to_opt_(nSize, return + 1) LPWSTR lpBuffer, _In_ uint32_t nSize);
-inline uint32_t PalGetEnvironmentVariable(_In_opt_ LPCWSTR lpName, _Out_writes_to_opt_(nSize, return + 1) LPWSTR lpBuffer, _In_ uint32_t nSize)
-{
-    return GetEnvironmentVariableW(lpName, lpBuffer, nSize);
-}
-#else
 _Success_(return != 0 && return < nSize)
 extern "C" uint32_t __stdcall GetEnvironmentVariableA(_In_opt_ LPCSTR lpName, _Out_writes_to_opt_(nSize, return + 1) LPSTR lpBuffer, _In_ uint32_t nSize);
-inline uint32_t PalGetEnvironmentVariable(_In_opt_ LPCSTR lpName, _Out_writes_to_opt_(nSize, return + 1) LPSTR lpBuffer, _In_ uint32_t nSize)
+inline uint32_t PalGetEnvironmentVariableA(_In_opt_ LPCSTR lpName, _Out_writes_to_opt_(nSize, return + 1) LPSTR lpBuffer, _In_ uint32_t nSize)
 {
     return GetEnvironmentVariableA(lpName, lpBuffer, nSize);
```

Comment:
`GetEnvironmentVariableA` has more overhead than `GetEnvironmentVariableW` since it needs to do the current code page conversions both ways. Also, switching to `GetEnvironmentVariableA` on Windows is going to expose us to Windows current code page mess that is its own ball of mud. It is not a problem at the moment because we seems to be only reading numbers at the moment, but it will be a problem if we ever need to start reading strings.

Reply:
I think it would be best to avoid calling current-code-page Win32 methods. 

It is fine with me to get rid of TCHAR and deal with the platform differences using some kind of different scheme.

Reply:
Doh, right. I got a bit overzealous - I mainly wanted to get rid of the TCHAR from the exposed/consuming side. I'll rework it to try to hide it away and keep Windows avoiding current code page